src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -63,120 +133,1983 @@ ActorInit En_Kaizoku_InitVars = {
+        this->unk_2B8 = 0;
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+    }
+
+    if (curFrame >= 25.0f) {
+        player = GET_PLAYER(play);
+        if (this->subCamId == 0) {
+            if (!ActorCutscene_GetCanPlayNext(this->unk_2D6)) {
+                ActorCutscene_SetIntentToPlay(this->unk_2D6);
+                return;
+            }
+            ActorCutscene_StartAndSetUnkLinkFields(this->unk_2D6, &this->picto.actor);
+            this->subCamId = ActorCutscene_GetCurrentSubCamId(this->picto.actor.cutscene);
+        }
+
+        Math_Vec3f_Copy(&this->unk_3C4, &gZeroVec3f);
+        player->actor.world.pos.x = this->picto.actor.home.pos.x + 90.0f;
+        player->actor.world.pos.z = this->picto.actor.home.pos.z + 30.0f;
+        this->picto.actor.world.pos.x = this->picto.actor.home.pos.x;
+        this->picto.actor.world.pos.z = this->picto.actor.home.pos.z;
 
-extern DamageTable D_80B8AB3C;
-extern ColliderCylinderInit D_80B8AB7C;
-extern ColliderQuadInit D_80B8ABA8;
+        this->subCamEye.x = this->unk_5C8.x = player->actor.world.pos.x + 39.0f;
+        this->subCamEye.y = this->unk_5C8.y = player->actor.world.pos.y + 4.0f;
+        this->subCamEye.z = this->unk_5C8.z = player->actor.world.pos.z - 41.0f;
 
-extern UNK_TYPE D_060058B8;
+        this->subCamAt.x = this->unk_5D4.x = player->actor.world.pos.x - 150.0f;
+        this->subCamAt.y = this->unk_5D4.y = player->actor.world.pos.y + 60.0f;
+        this->subCamAt.z = this->unk_5D4.z = player->actor.world.pos.z + 50.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kaizoku/EnKaizoku_Init.s")
+        player->actor.shape.rot.y = player->actor.world.rot.y =
+            Math_Vec3f_Yaw(&player->actor.world.pos, &this->picto.actor.world.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kaizoku/EnKaizoku_Destroy.s")
+        if (this->subCamId != 0) {
+            this->subCamUp.x = 0.0f;
+            this->subCamUp.y = 1.0f;
+            this->subCamUp.z = 0.0f;
+            Play_SetCameraAtEyeUp(play, this->subCamId, &this->subCamAt, &this->subCamEye, &this->subCamUp);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kaizoku/func_80B85858.s")
+    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.yawTowardsPlayer, 1, 0xFA0, 1);
+    SkelAnime_Update(&this->skelAnime);
+    if (this->frameCount <= curFrame) {
+        this->unk_2D8 = 0;
+        func_80B86B58(this);
+    } else if (Animation_OnFrame(&this->skelAnime, 10.0f)) {
+        Actor_PlaySfxAtPos(&this->picto.actor, NA_SE_EN_GERUDOFT_DOWN);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kaizoku/func_80B85900.s")
+void func_80B89A08(EnKaizoku* this, PlayState* play) {
+    s32 sp64 = 0;
+    Vec3f sp58;
+    s32 i;
+    Vec3f sp48;
+    Player* player;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kaizoku/func_80B85A00.s")
+    if (gSaveContext.save.playerData.health <= 16) {
```

Comment:
```suggestion
    if (gSaveContext.save.playerData.health <= 0x10) {
```
Huh, so they can't kill you.